DXRModules/DeusEx/Classes/DXREvents.uc

```diff
@@ -1959,98 +1959,139 @@ function string RemapBingoEvent(string eventname)
 
 }
 
-static function string GetBingoFailedGoals(DXRando dxr, string eventname, out string failed2)
+static function GetBingoFailedGoals(DXRando dxr, string eventname, out string failed[5])
 {
-    failed2 = "";
+    local int i;
+
+    for (i = 0; i < ArrayCount(failed); i++) {
+        failed[i] = "";
```

Comment:
if you instead return the number of items in the array, then you don't need to pre-clear the array, just overwrite and return the length

maybe use the variable name `i` or `num`, then each assignment would be like `failed[num++] = "LebedevLived";` then at the end you do `return num;`

Reply:
I'd much, much rather give a correct array of failures than trust whatever calls the function to also deal with the number of items it's told to.

Reply:
well you can leave it, but I still think returning the length is nice